sections/collage.liquid

```diff
@@ -39,12 +41,45 @@
                   {% endif %}
                 >
                   {%- if block.settings.image != blank -%}
-                    {%- capture sizes -%}(min-width: {{ settings.page_width }}px) {% if section.blocks.size == 1 %}calc({{ settings.page_width }}px - 100px){% else %}{{ settings.page_width | minus: 100 | times: 0.67 | round }}px{% endif %}, (min-width: 750px){% if section.blocks.size == 1 %} calc(100vw - 100px){% else %} 500px{% endif %}, calc(100vw - 30px){%- endcapture -%}
-                    {{
-                      block.settings.image
-                      | image_url: width: 3000
-                      | image_tag: loading: 'lazy', sizes: sizes, widths: '550, 720, 990, 1100, 1500, 2200, 3000'
-                    }}
+                    {%- liquid
+                      if section.settings.desktop_layout == 'left'
+                        assign large_block = forloop.first
+                      else
+                        assign large_block = forloop.last
+                      endif
+
+                      assign grid_space_desktop = settings.spacing_grid_horizontal | divided_by: 2 | append: 'px'
+                      assign grid_space_mobile = settings.spacing_grid_horizontal | divided_by: 4 | append: 'px'
```

Comment:
One thing I'm wondering about. Why does it need to be divided by 2 and 4 🤔 Since the spacing is applied once

Reply:
Because let's say there is a 40px gap, this 40px approximately equally split between 2 images. Since we are trying to calculate an image size we need to account it partially. If I account 40px for the first image only then the second image size will be calculated wrong.

Reply:
Makes sense 👍